### ISSUE TITLE
[FEAT] Graphql - configurable mutation InputType

### ISSUE THREAD

**Issue by florent-andre** [refinedev/refine#3409]
### Is your feature request related to a problem? Please describe.

In `create` and `update` mutation queries, `operation` name can be configured via `metadata`. `variable.input.type` can't be configured as it's "hard coded". 

There is case (or at list mine ;)), where `InputType.type` can't be the same as the "hardcoded by refine" ones.

### Describe the thing to improve

For example, in the [dataProvider.create](https://github.com/refinedev/refine/blob/master/packages/graphql/src/dataProvider/index.ts#L125), and in the same spirit that allow to configure `operation` : 
```
const operation = metaData?.operation ?? camelCreateName;
```
The variables.input.type should be something like: 
```
const variableInputType = metaData?.variableInputType ?? `${camelCreateName}Input`
```

What do you think about that ? Is it doable ? 
Thanks for this great piece of lib !

**Comment by yildirayunlu** [comment]
Hey @florent-andre 

GraphQL data provider needs various arrangements. This includes what you wrote. Can you create PR to support us? 🙏🏼

**Comment by florent-andre** [comment]
Yes, sure I can try (maybe this week or later on). Do you think `metaData?.variableInputType` is a good name ? 
Regards

**Comment by yildirayunlu** [comment]
> Yes, sure I can try (maybe this week or later on). Do you think `metaData?.variableInputType` is a good name ? Regards

I think `metaData?.variableInputType` can be found better, we are open to suggestions. We look forward to PR 🎉

**Comment by mattbho** [comment]
Hello! I'll open a PR for this one!
